src/kmpmetadata.cpp

```diff
@@ -81,11 +82,8 @@ KmpMetadata::KmpMetadata(int fd) {
         }
     }
 
-    if (auto *kmpKeyboards = json_object_object_get(obj.get(), "keyboards");
-        kmpKeyboards && json_object_get_type(kmpKeyboards) == json_type_array) {
-        for (size_t i = 0, e = json_object_array_length(kmpKeyboards); i < e;
-             i++) {
-            auto *file = json_object_array_get_idx(kmpKeyboards, i);
+    if (obj.contains("keyboards") && obj["keyboards"].is_array()) {
```

Comment:
can you avoid look up "keyboards" multiple times? Same to the other places in the file.